config/matrix-api-v1.yml

```diff
@@ -0,0 +1,299 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixResponse'
+        '404':
+          description: "Matrix request id not found."
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixErrorResponse'
+  /filters:
+    get:
+      summary: Get a list of permitted filters
+      description: >
+        Get a list of filters that a POST to /matrix will accept.
+      responses:
+        '200':
+          description: 'Successfully return filter list.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixFilters'
+  /filters/{filter_name}:
+    get:
+      summary: 'Get a description of a filter'
+      description: >
+        Get detailed information about a filter that can be applied
+        to the HCA expression data.
+      parameters:
+        - name: filter_name
+          schema:
+            type: string
+          required: True
+          in: path
+          description: 'Filter to describe.'
+      responses:
+        '200':
+          description: "Filter found."
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/NumericFieldDetail'
+                  - $ref: '#/components/schemas/CategoricalFieldDetail'
+        '404':
+          description: 'Filter not found.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixErrorResponse'
+  /fields:
+    get:
+      summary: Get a lsit of available metadata fields
+      description: >
+        Get a list of metadata field names that a POST to 
+        /matrix will accept.
+      responses:
+        '200':
+          description: 'Successfully return field list.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixFields'
+  /fields/{field_name}:
+    get:
+      summary: 'Get a description of a metadata field'
+      description: >
+        Get detailed information about a an available metadata field.
+      parameters:
+        - name: field_name
+          schema:
+            type: string
+          required: True
+          in: path
+          description: 'Field to describe.'
+      responses:
+        '200':
+          description: "Field found."
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/NumericFieldDetail'
+                  - $ref: '#/components/schemas/CategoricalFieldDetail'
+        '404':
+          description: 'Field not found.'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/MatrixErrorResponse'
+components:
+  schemas:
+    FilterOperation:
```

Comment:
I was reviewing Genevieve's [UX Summary](https://docs.google.com/document/d/1cZGlyXzzFCRuAJqcsh58VHQD1apjzAULa_7xQSL60oE/edit#) to help validate the API - avoiding cases that are more query API specific, would we want to support something like _For all pancreas cells from women age 40-50, return all cells where at least 500 genes have non-zero values, **or** total reads > X, etc_.

GDC supports a [broader range](https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query) of operators - including  _and_ and _or_. Is the intent to keep operations minimal with _AND_ as the default until further feedback is received?

Reply:
I had thought to start with _AND_. If we add _OR_, then we need to encode groupings too. That said, if it's already in the UX docs, we probably should just bite the bullet now.